src/vtp/web/api/backend.py

```diff
@@ -0,0 +1,219 @@
+        """Mock only - return a static cast ballot"""
+        with open(VtpBackend._MOCK_CAST_BALLOT, "r", encoding="utf8") as infile:
+            json_doc = json.load(infile)
+        return json_doc
+
+    @staticmethod
+    def mock_get_ballot_check() -> tuple[list, int]:
+        """Mock only - return a static cast ballot"""
+        with open(VtpBackend._MOCK_BALLOT_CHECK, "r", encoding="utf8") as infile:
+            csv_doc = list(csv.reader(infile))
+        return csv_doc, VtpBackend._MOCK_VOTER_INDEX
+
+    @staticmethod
+    def cast_ballot(vote_store_id: str, cast_ballot: dict) -> dict:
+        """
+        Endpoint #3: will cast (upload) a cast ballot and return the
+        ballot-check and voter-index
+        """
+        if VtpBackend._MOCK_MODE:
+            # Just return a mock ballot-check and voter-index
+            return VtpBackend.mock_get_ballot_check()
+        # handle the incoming ballot and return the ballot-check and voter-index
+        operation = AcceptBallotOperation(
+            election_data_dir=Common.get_guid_based_edf_dir(vote_store_id),
+        )
+        return operation.run(
+            cast_ballot_json=cast_ballot,
+            merge_contests=True,
+        )
+
+    @staticmethod
+    def verify_ballot_check(
+        vote_store_id: str,
+        ballot_check: list,
+        vote_index: int,
+        cvr: bool = False,
+    ) -> str:
+        """
+        Endpoint #4: will verify a ballot-check and vote-inded, returning an
+        undefined string at this time.
+        """
+        if VtpBackend._MOCK_MODE:
+            # Just return a mock verify ballot string
+            with open(VtpBackend._MOCK_VERIFY_BALLOT_LOG, "r", encoding="utf8") as infile:
+                json_doc = json.load(infile)
+            return json_doc
+        # handle the incoming ballot and return the ballot-check and voter-index
+        operation = VerifyBallotReceiptOperation(
+            election_data_dir=Common.get_guid_based_edf_dir(vote_store_id),
+            stdout_printing=False,
+        )
+        return operation.run(
+            receipt_data=ballot_check,
+            row=str(vote_index),
+            cvr=cvr,
+        )
+
+    @staticmethod
+    def tally_election_check(
+        vote_store_id: str,
+        contests: str,
+        digests: str,
+    ) -> str:
+        """
+        Endpoint #5: will tally an election and print stuff
+        """
+        if VtpBackend._MOCK_MODE:
+            # Just return a mock tally string
+            with open(VtpBackend._MOCK_TALLY_CONTESTS_LOG, "r", encoding="utf8") as infile:
+                json_doc = json.load(infile)
+            return json_doc
+        # handle the incoming ballot and return the ballot-check and voter-index
+        operation = TallyContestsOperation(
+            election_data_dir=Common.get_guid_based_edf_dir(vote_store_id),
+            stdout_printing=False,
+        )
+        if digests == "None":
```

Comment:
Correct.  Open to any other way of doing this - I do not know what is idiomatic here.

Reply:
The idiomatic way to show a value that's undefined is to use the `None` constant.

```suggestion
        if digests is None:
```

Reply:
@ion-oset Question - since digests is coming in from the endpoint and is (currently) delimited by "/"s, what is the proper way to specify no value at the endpoint?  Currently from the backend point of view digests can be nothing, a single git digest, or a comma separated list of git digests.  That is why I picked '"None"' as a place holder to figuring out what the correct idiom is.